src/Connection/Connection.php

```diff
@@ -346,18 +347,20 @@ public function beginTransaction($isolationLevel = null): Transaction
      * {@see queryCache}
      * {@see noCache()}
      */
-    public function cache(callable $callable, ?int $duration = null, ?Dependency $dependency = null)
+    public function cache(callable $callable, int $duration = null, Dependency $dependency = null)
     {
-        $this->queryCacheInfo[] = [$duration ?? $this->queryCacheDuration, $dependency];
+        $this->queryCache->setCacheInfo(
+            [$duration ?? $this->queryCache->getCacheDuration(), $dependency]
+        );
 
         try {
             $result = $callable($this);
 
-            array_pop($this->queryCacheInfo);
+            $this->queryCache->cacheInfoArrayPop();
```

Comment:
`finally` could be used. Then it will be just try - finally without catch.

Reply:
done.